frontend/src/routes/(authenticated)/project/create/+page.svelte

```diff
@@ -68,6 +63,19 @@
       await goto('/');
     }
   });
+
+  const asyncCodeError = writable<string | undefined>();
+  const { debounce: validateCode } = makeAsyncDebouncer(
+    () => {
+      const code = $form.code;
+      if (!code || !user.canCreateProjects) return Promise.resolve(true);
+      return _projectCodeAvailable(code);
+    },
+    (success) => asyncCodeError.set(success ? undefined : $t('project.create.code_exists')),
```

Comment:
looking at this I wonder if it would make more sense to debounce a derived async store, where the value is the result of the async function being called with the store passed in, in this case the code is passed in, and there's an async callback which is set on the store which is the return value of the make function. Does that make sense? If not feel free to ignore it and leave it like this.